pep-0458.txt

```diff
@@ -542,38 +542,49 @@ that disagrees with what is available on PyPI, which is indistinguishable from
 arbitrary metadata injected by an attacker.  The problem would also occur for
 mirrors attempting to sync with PyPI.
 
-
 Consistent Snapshots
 --------------------
 
-There are problems with consistency on PyPI with or without TUF.  TUF requires
-that its metadata be consistent with the repository files, but how would the
-metadata be kept consistent with projects that change all the time?  As a
-result, this proposal MUST address the problem of producing a consistent
-snapshot that captures the state of all known projects at a given time.  Each
-snapshot should safely coexist with any other snapshot, and be able to be
-deleted independently, without affecting any other snapshot.
+To keep TUF metadata on PyPI consistent with the highly volatile target files,
+consistent snapshots SHOULD be used. Each consistent snapshot captures the
+state of all known projects at a given time and MAY safely coexist with any
+other snapshot, or be deleted independently, without affecting any other
+snapshot.
+
+To maintain consistent snapshots, all TUF metadata MUST, when written to disk,
+include a version number in their filename:
 
-The solution presented in this PEP is that every metadata or data file managed
-by PyPI and written to disk MUST include in its filename the `cryptographic
-hash`__ of the file.  How would this help clients that use the TUF protocol to
-securely and consistently install or update a project from PyPI?
+  VERSION_NUMBER.ROLENAME.json,
+    where VERSION_NUMBER is an incrementing integer, and ROLENAME is one of the
+    top-level metadata roles -- *root*, *snapshot* or *targets* -- or one of
+    the delegated targets roles -- *bins* or *bin <i>*.
```

Comment:
Elsewhere, I think we refer to the delegated bins roles as *bin-n*. I don't have a particular preference, but I think we should be consistent.

Reply:
I agree. I think it's not fully consistent elsewhere either. We should, probably in a separate PR, go through the entire document and clearly disambiguate between *bins* and *bin-n* roles/metadata.